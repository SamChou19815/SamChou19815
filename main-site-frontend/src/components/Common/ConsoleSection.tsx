import React, { ReactElement, ReactNode } from 'react';
import styles from './ConsoleSection.module.css';

type Props = {
  readonly id: string;
  readonly title: string;
  readonly children: ReactNode;
};

export default ({ id, title, children }: Props): ReactElement => (
  <section id={id} className={styles.Section}>
    <h3 className={styles.Title}>
      <code>
        {'> '}
        {title}
      </code>
    </h3>
    {children}
  </section>
);
