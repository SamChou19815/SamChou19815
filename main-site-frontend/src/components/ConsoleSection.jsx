// @flow strict

import type { Node } from 'react';
import React from 'react';
import styles from './ConsoleSection.module.css';

type Props = {|
  +id: string;
  +title: string;
  +children: Node;
|};

export default ({ id, title, children }: Props): Node => (
  <section id={id} className={styles.Section}>
    <h3 className={styles.Title}>
      <code>
        {'> '}
        {title}
      </code>
    </h3>
    {children}
  </section>
);
