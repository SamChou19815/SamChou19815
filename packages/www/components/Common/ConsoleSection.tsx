import React, { ReactElement, ReactNode } from 'react';

import styles from './ConsoleSection.module.css';

type Props = {
  readonly id: string;
  readonly title: string;
  readonly className?: string;
  readonly titleClassName?: string;
  readonly children: ReactNode;
};

const classes = (firstClass: string, secondClass?: string): string =>
  secondClass === undefined ? firstClass : `${firstClass} ${secondClass}`;

export default ({ id, title, className, titleClassName, children }: Props): ReactElement => (
  <section id={id} className={classes(styles.Section, className)}>
    <h3 className={classes(styles.Title, titleClassName)}>
      <code>
        $&nbsp;
        {title}
      </code>
    </h3>
    {children}
  </section>
);
