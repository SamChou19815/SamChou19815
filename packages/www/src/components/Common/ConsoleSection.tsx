import clsx from 'clsx';
import type { ReactElement, ReactNode } from 'react';

import styles from './ConsoleSection.module.css';

type Props = {
  readonly id: string;
  readonly title: string;
  readonly className?: string;
  readonly titleClassName?: string;
  readonly children: ReactNode;
};

const ConsoleSection = ({
  id,
  title,
  className,
  titleClassName,
  children,
}: Props): ReactElement => (
  <section id={id} className={className}>
    <h3 className={clsx(styles.Title, titleClassName)}>
      <code>
        $&nbsp;
        {title}
      </code>
    </h3>
    {children}
  </section>
);

export default ConsoleSection;
