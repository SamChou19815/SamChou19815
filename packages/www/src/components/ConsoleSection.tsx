import React, { ReactNode } from 'react';

type Props = {
  readonly id: string;
  readonly title: string;
  readonly className?: string;
  readonly titleClassName?: string;
  readonly children: ReactNode;
};

const ConsoleSection = ({ id, title, className, titleClassName, children }: Props): JSX.Element => (
  <section id={id} className={className}>
    <h3
      className={
        titleClassName == null ? 'console-section-title' : `console-section-title ${titleClassName}`
      }
    >
      <code>
        $&nbsp;
        {title}
      </code>
    </h3>
    {children}
  </section>
);

export default ConsoleSection;
