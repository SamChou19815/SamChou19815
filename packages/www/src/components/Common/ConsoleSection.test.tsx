import React from 'react';
import renderer from 'react-test-renderer';
import ConsoleSection from './ConsoleSection';

it('ConsoleSection matches snapshot.', () => {
  const tree = renderer
    .create(
      <ConsoleSection id="test" title="title">
        A
      </ConsoleSection>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
