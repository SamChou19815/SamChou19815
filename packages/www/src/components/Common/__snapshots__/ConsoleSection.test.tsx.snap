// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleSection matches snapshot. 1`] = `
<section
  className="Section"
  id="test"
>
  <h3
    className="Title"
  >
    <code>
      &gt; 
      title
    </code>
  </h3>
  A
</section>
`;
